src/java.net.http/share/classes/java/net/http/HttpClient.java

```diff
@@ -354,16 +355,60 @@ public static Builder newBuilder() {
          */
         public Builder authenticator(Authenticator authenticator);
 
+        /**
+         * Binds the socket to this local address when creating
+         * connections for sending requests.
+         *
+         * <p> If no local address is set or {@code null} is passed
+         * to this method then sockets created by the
+         * HTTP client will be bound to an automatically
+         * assigned socket address.
+         *
+         * <p> Common usages of the {@code HttpClient} do not require
+         * this method to be called. Setting a local address, through this
+         * method, is only for advanced usages where users of the {@code HttpClient}
+         * require specific control on which network interface gets used
+         * for the HTTP communication. Callers of this method are expected to
+         * be aware of the networking configurations of the system where the
+         * {@code HttpClient} will be used and care should be taken to ensure the
+         * correct {@code localAddr} is passed. Failure to do so can result in
+         * requests sent through the {@code HttpClient} to fail.
+         *
+         * @implSpec The default implementation of this method throws
+         * {@code UnsupportedOperationException}. {@code Builder}s obtained
+         * through {@link HttpClient#newBuilder()} provide an implementation
+         * of this method that allows setting the local address.
+         *
+         * @param localAddr The local socket address. Can be null.
+         * @return this builder
+         * @throws UnsupportedOperationException if this builder doesn't support
+         *         configuring a local address
```

Comment:
Maybe there should be an `@implNote` to say that the default implementation of the HttpClient only supports `InetSocketAddress` - and that the port should be 0?
Possibly also extend the `@throws UnsupportedOperationException` to cover the case where the given address is not supported?
Something like:

```
         * @throws UnsupportedOperationException if this builder doesn't support
         *         configuring a local address, or if the given local address is not one
         *         supported by the concrete {@code HttpClient} implementation.
```

Reply:
Hello Daniel,
I have updated the PR to include a new `@implNote` whose text is similar to what you suggest here.